docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+
+Be aware however, that there are some limitations that we discussed with the Kubernetes networking SIG and [Tim Hockin](https://www.linkedin.com/in/tim-hockin-6501072) specifically and that the maintainers are reluctant to drop. Those are called [safeguards](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/#safeguards) and if they strike, the hints are off and traffic is routed again randomly. Especially controversial is the balancing limitation as there is the assumption, that the load that hits an endpoint is determined by the allocatable CPUs in that topology zone, but that's not always if even mostly the case (we opened among others [#110714](https://github.com/kubernetes/kubernetes/issues/110714)). So, this limitation hits far too often and your hints are off, but then again, it's about network latency and cost optimization first, so it's better than nothing.
+
+## Networking
+
+We have talked about networking only to some small degree so far (`readiness` probes, pod disruption budgets, topology aware hints). The most important component is probably your ingress load balancer - everything else is managed by Kubernetes. AWS, Azure, GCP, and also OpenStack offer multi-zonal load balancers, so make use of them. In Azure and GCP, LBs are regional whereas in AWS and OpenStack, they need to be bound to a zone, which the cloud-controller-manager does by observing the zone labels at the nodes (please note that this behavior is not always working as expected, see [#570](https://github.com/kubernetes/cloud-provider-aws/issues/569) that we opened on the AWS cloud-controller-manager not readjusting to newly observed zones).
+
+Please be reminded that even if you use a service mesh like [Istio](https://istio.io), the off-the-shelf installation/configuration usually never comes with productive qualities and you will have to fine-tune your installation/configuration, much like the rest of your workload.
+
+## Relevant Cluster Settings
+
+Following now a summary/list of the more relevant settings you may like to tune for Gardener-managed clusters:
+
+``` yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+spec:
+  controlPlane:
+    highAvailability:
+      failureTolerance:
+        type: zone # valid values are `node` and `zone` (only available if your control plane resides in a region with 3+ zones)
+  kubernetes:
+    kubeAPIServer:
+      defaultNotReadyTolerationSeconds: 300
+      defaultUnreachableTolerationSeconds: 300
+    kubelet:
+      ...
+    kubeScheduler:
+      featureGates:
+        MinDomainsInPodTopologySpread: true
+    kubeControllerManager:
+      nodeMonitorPeriod: 10s
+      nodeMonitorGracePeriod: 2m0s
```

Comment:
We're now at the point where we're tuning these example properties in a doc, because we know what happens: people will copy them verbatim to their yamls. That's what I wanted to avoid, unless we have good reasons to provide these copy-paste snippets or want to point to very specific properties.

Reply:
`40s` is the default value ([ref](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)), no fine tuning. I don't know why we explicitly wanted to point to `2m0s` but maybe Vedran can explain.

Reply:
@timuthy I didn't want to point to or recommend 2m. I copied the fields from a shoot cluster for illustration. It's the default in gardener (2m, not 40s). The reason is somewhat convoluted. I changed it now to 40s, but you will know that today, this makes DWD pointless as it will then react too late.

Reply:
Indeed, we are in a miserable situation with this setting at the moment and maybe I added the result of our latest offline discussion too early here.